Classes/Replicator/NodeReplicator.php

```diff
@@ -50,6 +50,18 @@ public function similarizeNodeVariants(NodeInterface $node): void
         /** @var NodeInterface $parentVariant */
         foreach ($this->getParentVariants($node) as $parentVariant) {
             $nodeVariant = $parentVariant->getContext()->getNodeByIdentifier($node->getIdentifier());
+
+            if(!$nodeVariant instanceof NodeInterface) {
+                $dimensionsAndPresets = [];
+                foreach ($parentVariant->getDimensions() as $dimension => $presets) {
+                    $dimensionsAndPresets[] = $dimension . ':' . implode(',', array_values($presets));
+                }
+                $dimensionString = implode('|', $dimensionsAndPresets);
+
+                $this->logger->info(sprintf('[ParentNodeIdentifier: %s, TargetDimension: %s] %s', $parentVariant->getIdentifier(), $dimensionString, 'Cannot similarize node in, as the node was not found in that dimension.'), LogEnvironment::fromMethodName(__METHOD__));
```

Comment:
Why not log the actual nodeIdentifier instead of the parent?
```suggestion
                $this->logger->info(sprintf('[NodeIdentifier: %s, TargetDimension: %s] Cannot similarize node, as the node was not found in the target dimension.', $node->getIdentifier(), $dimensionString), LogEnvironment::fromMethodName(__METHOD__));
```